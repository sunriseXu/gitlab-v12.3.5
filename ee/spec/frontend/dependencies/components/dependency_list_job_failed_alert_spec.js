import { createLocalVue, shallowMount } from '@vue/test-utils';
import { GlButton } from '@gitlab/ui';
import DependencyListAlert from 'ee/dependencies/components/dependency_list_alert.vue';
import DependencyListJobFailedAlert from 'ee/dependencies/components/dependency_list_job_failed_alert.vue';

describe('DependencyListJobFailedAlert component', () => {
  let wrapper;

  const factory = (options = {}) => {
    const localVue = createLocalVue();

    wrapper = shallowMount(localVue.extend(DependencyListJobFailedAlert), {
      localVue,
      sync: false,
      ...options,
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  it('matches the snapshot', () => {
    factory({ propsData: { jobPath: '/jobs/foo/3210' } });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('inludes a button button if "jobPath" is given', () => {
    factory({ propsData: { jobPath: '/jobs/foo/3210' } });

    expect(wrapper.find(GlButton).exists()).toBe(true);
  });

  it('does not include a button if "jobPath" is not given', () => {
    factory();

    expect(wrapper.find(GlButton).exists()).toBe(false);
  });

  it.each([undefined, null, ''])(
    'does not include a button if "jobPath" is given but empty',
    jobPath => {
      factory({ propsData: { jobPath } });

      expect(wrapper.find(GlButton).exists()).toBe(false);
    },
  );

  describe('when the generic alert component emits a close event', () => {
    let closeListenerSpy;

    beforeEach(() => {
      closeListenerSpy = jest.fn();

      factory({
        propsData: { jobPath: '/jobs/foo/3210' },
        listeners: {
          close: closeListenerSpy,
        },
      });

      wrapper.find(DependencyListAlert).vm.$emit('close');
    });

    it('calls the given listener', () => {
      expect(closeListenerSpy).toHaveBeenCalledTimes(1);
    });
  });
});
