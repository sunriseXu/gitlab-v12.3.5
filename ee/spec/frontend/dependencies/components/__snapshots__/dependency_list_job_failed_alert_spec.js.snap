// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DependencyListJobFailedAlert component matches the snapshot 1`] = `
<dependencylistalert-stub
  headertext="Job failed to generate the dependency list"
  type="danger"
>
  <p>
    The 
    <code>
      dependency_scanning
    </code>
     job has failed and cannot generate the list. Please ensure the job is running properly and run the pipeline again.
  </p>
   
  <glbutton-stub
    class="btn-inverted btn-danger mb-2"
    href="/jobs/foo/3210"
  >
    
    View job
  
  </glbutton-stub>
</dependencylistalert-stub>
`;
